compiler/src/iree/compiler/Dialect/Flow/Transforms/test/set_encoding.mlir

```diff
@@ -1,240 +1,237 @@
     %arg2 : tensor<100x500xf32>) -> tensor<100x500xf32> {
   %0 = linalg.matmul ins(%arg0, %arg1 : tensor<100x250xf32>, tensor<250x500xf32>)
       outs(%arg2 : tensor<100x500xf32>) -> tensor<100x500xf32>
   return %0 : tensor<100x500xf32>
 }
-//      CHECK: func @matmul_padding(
+//      CHECK: func @matmul_f32f32f32(
 // CHECK-SAME:     %[[ARG0:.+]]: tensor<100x250xf32>
 // CHECK-SAME:     %[[ARG1:.+]]: tensor<250x500xf32>
 // CHECK-SAME:     %[[ARG2:.+]]: tensor<100x500xf32>
-//      CHECK:   %[[LHS_PAD:.+]] = tensor.pad %[[ARG0]] low[0, 0] high[12, 6]
-//      CHECK:       tensor<100x250xf32> to tensor<112x256xf32>
-//      CHECK:   %[[RHS_PAD:.+]] = tensor.pad %[[ARG1]] low[0, 0] high[6, 12]
-//      CHECK:       tensor<250x500xf32> to tensor<256x512xf32>
-//      CHECK:   %[[OUTS_PAD:.+]] = tensor.pad %[[ARG2]] low[0, 0] high[12, 12]
-//      CHECK:       tensor<100x500xf32> to tensor<112x512xf32>
+//      CHECK:   %[[LHS_PADDING_SIZE:.+]]:2 = iree_linalg_ext.encoding_padding_size %[[ARG0]] : tensor<100x250xf32> <user = MATMUL_F32F32F32, role = LHS> -> index, index
+//      CHECK:   %[[LHS_PAD:.+]] = tensor.pad %[[ARG0]] low[0, 0] high[%[[LHS_PADDING_SIZE]]#0, %[[LHS_PADDING_SIZE]]#1]
+//      CHECK:       tensor<100x250xf32> to tensor<?x?xf32>
 //      CHECK:   %[[LHS:.+]] = iree_linalg_ext.set_encoding %[[LHS_PAD]]
-// CHECK-SAME:       tensor<112x256xf32, #iree_linalg_ext.encoding<user = MATMUL_F32F32F32, role = LHS>>
+// CHECK-SAME:       tensor<?x?xf32, #iree_linalg_ext.encoding<user = MATMUL_F32F32F32, role = LHS, orig_type = tensor<100x250xf32>>>
+//      CHECK:   %[[RHS_PADDING_SIZE:.+]]:2 = iree_linalg_ext.encoding_padding_size %[[ARG1]] : tensor<250x500xf32> <user = MATMUL_F32F32F32, role = RHS> -> index, index
+//      CHECK:   %[[RHS_PAD:.+]] = tensor.pad %[[ARG1]] low[0, 0] high[%[[RHS_PADDING_SIZE]]#0, %[[RHS_PADDING_SIZE]]#1]
+//      CHECK:       tensor<250x500xf32> to tensor<?x?xf32>
 //      CHECK:   %[[RHS:.+]] = iree_linalg_ext.set_encoding %[[RHS_PAD]]
-// CHECK-SAME:       tensor<256x512xf32, #iree_linalg_ext.encoding<user = MATMUL_F32F32F32, role = RHS>>
+// CHECK-SAME:       tensor<?x?xf32, #iree_linalg_ext.encoding<user = MATMUL_F32F32F32, role = RHS, orig_type = tensor<250x500xf32>>>
+//      CHECK:   %[[OUTS_PADDING_SIZE:.+]]:2 = iree_linalg_ext.encoding_padding_size %[[ARG2]] : tensor<100x500xf32> <user = MATMUL_F32F32F32, role = RESULT> -> index, index
+//      CHECK:   %[[OUTS_PAD:.+]] = tensor.pad %[[ARG2]] low[0, 0] high[%[[OUTS_PADDING_SIZE]]#0, %[[OUTS_PADDING_SIZE]]#1]
+//      CHECK:       tensor<100x500xf32> to tensor<?x?xf32>
 //      CHECK:   %[[OUTS:.+]] = iree_linalg_ext.set_encoding %[[OUTS_PAD]]
-// CHECK-SAME:       tensor<112x512xf32, #iree_linalg_ext.encoding<user = MATMUL_F32F32F32, role = RESULT>>
+// CHECK-SAME:       tensor<?x?xf32, #iree_linalg_ext.encoding<user = MATMUL_F32F32F32, role = RESULT, orig_type = tensor<100x500xf32>>>
```

Comment:
can you expand this to `original_type`? (abbreviations aren't generally very friendly to those without familiarity with the context or language)